CONTRIBUTING.rst

```diff
@@ -49,34 +45,92 @@ remember that SimPEG is maintained on a volunteer basis. If you suggest an
 enhancement, we certainly appreciate if you are also willing to take action
 and start a pull request!
 
+.. _questions:
+
+Questions
+---------
+
+If you have a question regarding a specific use of SimPEG, the fastest way
+to get a response is by posting on our Discourse discussion forum:
+https://simpeg.discourse.group/. Alternatively, if you prefer real-time chat,
+you can join our slack group at http://slack.simpeg.xyz.
+Please do not create an issue to ask a question.
+
+.. _pull_requests:
+
+Pull Requests
+=============
+
+We welcome contributions to SimPEG in the form of pull requests (PR)
 
 .. _contributing_new_code:
 
 Contributing new code
 ---------------------
 
+.. _getting started: https://docs.simpeg.xyz/content/basic/installing_for_developers.html
+
+.. _practices: https://docs.simpeg.xyz/content/basic/practices.html
+
+.. _testing: https://docs.simpeg.xyz/content/basic/practices.html#testing
+
+.. _documentation: https://docs.simpeg.xyz/content/basic/practices.html#documentation
+
+.. _code style: https://docs.simpeg.xyz/content/basic/practices.html#style
+
 If you have an idea for how to improve SimPEG, please first create an issue
-and :ref:`suggest an enhancement <suggesting_enhancements>`. We will use the
+and `suggest enhancements`_. We will use the
 issue as a place to discuss and make decisions on the suggestion. Once you are
 ready to take action and commit some code to SimPEG, please check out
-:ref:`Getting Started for Developers <getting_started_developers>` for
-tips on setting up a development environment and :ref:`Practices <practices>`
+`getting started`_ for
+tips on setting up a development environment and `practices`_
 for a description of the development practices we aim to follow. In particular,
 
-- :ref:`testing <testing>`
-- :ref:`documentation <documentation>`
-- :ref:`code style <style>`
+- `testing`_
+- `documentation`_
+- `code style`_
 
 are aspects we look for in all pull requests. We do code reviews on pull
 requests, with the aim of promoting best practices and ensuring that new
 contributions can be built upon by the SimPEG community.
 
+.. _pr_stages:
+
+Stages of a pull request
+------------------------
+
+When first creating a pull request, try to make your suggested changes as tightly
+scoped as possible. The fewer changes you make, the faster your branch will be
+merged!
+
+If your pull request is not ready for final review, but you still want feedback
```

Comment:
```suggestion
If your pull request (PR) is not ready for final review, but you still want feedback
```

Reply:
I add the abbreviation a little higher up.